### PR TITLE
Remove add_subplant_id optional argument

### PR DESCRIPTION
**Pull request by rouille**
### Purpose
Clean and allocate generation and fuel in EIA-923 data at sub-plant level only. The option to do it at the plant level was bugged (see #360) and is now deactivated. Closes #360. Closes CAR-4207.

Add missing typehints and docstrings to the functions 

### What the code is doing
Remove `add_subplant_id` from functions and discard `if` logic. 

### Testing
Run 2012 pipeline successfully

### Where to look
The `oge.data_cleaning.clean_eia923` and `oge.data.create_primary_fuel_table` are the 2 functions that are modified.
Add missing typehints and docstrings to these functions.

### Usage Example/Visuals
N/A

### Review estimate
5min

### Future work
N/A

### Checklist
- [x] Update the documentation to reflect changes made in this PR
- [x] Format all updated python files using `black`
- [x] Clear outputs from all notebooks modified
- [x] Add docstrings and type hints to any new functions created

### CODE REVIEW

File: src/oge/data_cleaning.py [hunk 1]
```diff
@@ -49,12 +49,27 @@
...
+        year (int): year to consider.
+        small (bool): should a subset of data be considered.
+        calculate_nox_emissions (bool, optional): whether or not clculate NOx emission
+            from fuel consumption. Defaults to True.
+        calculate_so2_emissions (bool, optional): whether or not clculate So2 emission
+            from fuel consumption. Defaults to True.
+
+    Returns:
+        tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame]: the first data frame is the
+            generation and fuel alocated with NOx and SO2 emission if requested. The
```
Comment (grgmiller) [inline src/oge/data_cleaning.py:68]: this dataframe also has GHG emissions data (always), and only NOX and SO2 if reqeusted

File: src/oge/data_cleaning.py [hunk 2]
```diff
@@ -49,12 +49,27 @@
...
+            second data frame is the primary fuel table. Finally, the third data frame
+            encloses the emission factors. All three data frames are at the subplant
```
Comment (grgmiller) [inline src/oge/data_cleaning.py:70]: The third data frame has subplant-specific emission factors, which is used for filling missing emissions data in cems later in the pipeline

**Review by grgmiller — APPROVED** [review]
Looks good - just two small clarifications for the docstring

**Comment by grgmiller** [comment]
Also make sure to add "Closes CAR-XXXX" to the PR description to tag the linear issue

File: src/oge/data_cleaning.py (hunk 1 above)
Comment (rouille) [reply src/oge/data_cleaning.py]: Done

File: src/oge/data_cleaning.py (hunk 2 above)
Comment (rouille) [reply src/oge/data_cleaning.py]: Done

**Comment by rouille** [comment]
> Also make sure to add "Closes CAR-XXXX" to the PR description to tag the linear issue

Added.

**Review by grgmiller — APPROVED** [review]
Thanks, looks good

**Pull request merged**